docs/interfaces/protobuf/reference.md

```diff
@@ -223,141 +318,323 @@ Where:
+!!! detail "Identical to `parseArrayToDict` except the intermediate Protobuf message is created on a Google Arena"
 
     Can improve memory allocation performance for large messages with deep repeated fields/map.
-
+    
     :fontawesome-brands-google:
     [Google Arenas](https://developers.google.com/protocol-buffers/docs/reference/arenas)
 
 ```q
-q)data:(1 2i;10 20f;`s1`s2)
-q).protobufkdb.serializeArrayArena[`RepeatedExampleDynamic;data]
+q)fields:.protobufkdb.getMessageFields[`RepeatedExampleDynamic]
+q)data:fields!(1 2i;10 20f;("s1";"s2"))
+q)array:.protobufkdb.serializeArrayFromDict[`RepeatedExampleDynamic;data]
+q)array
 "\n\002\001\002\022\020\000\000\000\000\000\000$@\000\000\000\000\000\0004@\0..
+q).protobufkdb.parseArrayArenaToDict[`RepeatedExampleDynamic;array]
+rep_int32 | 1    2
+rep_double| 10   20
+rep_string| "s1" "s2"
 ```
 
+## `saveMessageFromList`
 
+_Serialize from a kdb+ mixed list object to a Protobuf message file_
 
+```txt
+.protobufkdb.saveMessageFromList[message_type;file_name;msg_in]
+```
 
-## `version`
+Where
 
-_Library version (integer) used by the interface_
+-   `message_type` is a message type (string or symbol) matching a message name in the `.proto` definition
+-   `file_name` is the name of a file (string or symbol)
+-   `msg_in` is a kdb+ mixed list object 
+
+the function 
+
+1.  converts `msg_in` to a Protobuf message of `message_type`, serializes it, and writes it to `file_name`
+2.  returns generic null
+
+```q
+q)data
+1    2
+10   20
+"s1" "s2"
+q).protobufkdb.saveMessageFromList[`RepeatedExampleDynamic;`trivial_message_file;data]
+```
+
+## `loadMessageToList`
+
+_Parse from a Protobuf message file to a kdb+ mixed list object_
 
 ```txt
-.protobufkdb.version[]
+.protobufkdb.loadMessageToList[message_type;file_name]
 ```
 
-returns the version of the `libprotobuf` shared object used by the interface, as an integer.
+Where:
+
+-   `message_type` is a message type (string or symbol) matching a message name in the `.proto` definition
+-   `file_name` is the name of a file (string or symbol)
+
+returns a kdb+ mixed list object parsed from `file_name`.
 
 ```q
-q).protobufkdb.version[]
-3012003i
+q)data
+1    2
+10   20
+"s1" "s2"
+q).protobufkdb.saveMessageFromList[`RepeatedExampleDynamic;`trivial_message_file;data]
+q).protobufkdb.loadMessageToList[`RepeatedExampleDynamic;`trivial_message_file]
+1    2
+10   20
+"s1" "s2"
 ```
 
+## `saveMessageFromDict`
 
-## `versionStr`
+_Serialize from a kdb+ dictionary to a Protobuf message file_
 
-_Library version (string) used by the interface_
+```txt
+.protobufkdb.saveMessageFromDict[message_type;file_name;msg_in]
+```
+
+Where
+
+-   `message_type` is a message type (string or symbol) matching a message name in the `.proto` definition
+-   `file_name` is the name of a file (string or symbol)
+-   `msg_in` is a kdb+ dictionary from field name symbols to field values
+
+the function 
```

Comment:
Need this line? For example loadMessage does not have this line.

Reply:
Again, I think 'the function' is used when the function's operations is described in multiple steps